account_reconciliation_widget/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -1603,7 +1603,10 @@ odoo.define("account.ReconciliationModel", function (require) {
                     );
                     if (matching && matching.length == 2) {
                         matching = matching[1].replace(
-                            new RegExp("\\D" + reconcileModel.decimal_separator, "g"),
+                            new RegExp(
+                                "[^-d" + reconcileModel.decimal_separator + "]",
+                                "g"
+                            ),
                             ""
                         );
                         matching = matching.replace(
```

Comment:
Why this second replace then?

Reply:
That's unrelated. It seems to me that discussing other areas of the original code and why it does does things the way it does them is a different question. If you think it's wrong, file an issue and submit a PR.

But this fix here is certainly correct. Let's get it merged!

Reply:
The second replace allow to internationalize the float number (by setting it's decimal separator to a dot)